README.md

```diff
@@ -1736,6 +1736,9 @@ The following extractors use this feature:
 * `skip`: One or more of `webpage` (skip initial webpage download), `authcheck` (allow the download of playlists requiring authentication when no initial webpage is downloaded. This may cause unwanted behavior, see [#1122](https://github.com/yt-dlp/yt-dlp/pull/1122) for more details)
 * `approximate_date`: Extract approximate `upload_date` and `timestamp` in flat-playlist. This may cause date-based filters to be slightly off
 
+#### generic
+* `hls_segment_query`: If a generic m3u8 URL has a query, have the native HLS downloader append the query to the HLS segment URLs. Does not apply to ffmpeg
```

Comment:
```suggestion
* `hls_segment_query`: Passthrough any query in mpd/m3u8 manifest URLs to their fragments. Does not apply to ffmpeg
```
See https://github.com/yt-dlp/yt-dlp/pull/5604#discussion_r1027252728

Reply:
should this be renamed to just `segment_query` or `fragment_query` if it's going to be the same arg used for HLS and DASH?

Reply:
Pls do

Reply:
also, when working on #5604, I encountered HLS media playlist URLs that required their master playlist's URL query passed to them. should I add functionality for that in this PR as well? and if so, should it be triggered by the same arg, or a different one, like `variant_query` or something

Reply:
hm.. Since we don't have any generic example of this, I wouldn't bother. But if you want to add it, I won't stop you either